onnxscript/rewriter/pattern.py

```diff
@@ -1591,6 +1673,37 @@ def _apply_to_graph_or_function(
                 # is sufficient for patterns with a single output-node "node", which can serve as the
                 # insertion-point.
                 onnxscript.optimizer.basic_constant_propagation(delta.new_nodes)
+                if rule.as_function:
+                    # Create a function out of a copy of the matched nodes
+                    if len(delta.new_nodes) != 1:
+                        raise ValueError(
+                            "as_function=True is only supported for patterns with a single replacement node."
```

Comment:
Curious why the limitation?

Reply:
I think some version of this note would be helpful as comments for future readers